### ISSUE TITLE
Filter Alerts for Notification 

### ISSUE THREAD

**Issue by btmolloy** [louislam/uptime-kuma#3585]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature, Other

### 🔖 Feature description

Uptime Kuma is an amazing replacement for Uptime Robot as Kuma excels in places Uptime Robot can not. However, there are a few features that I really wish Kuma had, such as filtering of alerts. What I mean by this is the ability to filter what alerts are sent on a notification type. For example, I have a ton of sites that I like to monitor with Kuma, and Id love to be able to send SSL/Cert Alerts to one google chat, and the rest of the alerts to another chat. This would allow more organization of messages and allow a more efficient workflow. My current fix for this is very much a duck tape type fix where I essentially duplicated the google chat notification code and made one of them have server logic that only allows SSL messages to be sent and the other does the opposite. 

### ✔️ Solution

A perfect solution for this would be some sort of checkbox or drop down on the Notification Setup menu that would allow for more configuration of what alerts will be sent through the notification method, (i.e. All, SSL/CERT ONLY, KEYWORD ONLY, 503 ERROR?). 
<img width="492" alt="Screenshot 2023-08-16 at 10 46 37 AM" src="https://github.com/louislam/uptime-kuma/assets/92553518/a0de51b3-1169-4933-ba05-f71ed6a34366">

### ❓ Alternatives

Im not sure of any alternative solutions for this issue. I have a duck tape type fix right now which can be seen in the attached image. **I am NOT at all suggesting this is a viable solution, this is only what I had time for.** 
<img width="478" alt="Screenshot 2023-08-16 at 10 52 26 AM" src="https://github.com/louislam/uptime-kuma/assets/92553518/d6f830d5-de2a-49c8-81a0-a937be17e7ff">

**Comment by CommanderStorm** [comment]
An alternative is to use [Grafana Oncall](https://grafana.com/products/cloud/oncall/) for your escalation needs (see #2783 => this will probably be avaliable in `v2.0`)

Oncall includes fully customisable routing of allerts
![image](https://github.com/louislam/uptime-kuma/assets/26258709/0d6fdce9-f68e-4aff-a50e-39071dc84e74)

**Comment by CommanderStorm** [comment]
> I'm curious if there is a more simple or primitive solution here

Currently not, but that is why you opened the `feature-request`. ^^

> Louislam is the one who determines if suggested feature ideas are going to be looked into any further right?

Yes, Louis has the final say on what gets merged.
If you'd like to implement such a feature, please 
1. device a plan what should be changed how to achieve the goal of an escalation system
    (filtering alerts to different channels is essentially a basic building block of an escalation system)
2. discuss said plan as laid out here: https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md
    (rationale: saving time on your + Louis end)
3. Change what needs to be changed
4. Wait for a review + merge/revisiting of missed topics

tldr: Reasonable implementations of reasonable features are likely to get merged.

**Comment by CommanderStorm** [comment]
Related #1315 

Note that this feature-request is also a duplicate of https://github.com/louislam/uptime-kuma/issues/1233 and partially https://github.com/louislam/uptime-kuma/issues/1137
